genesyscloud/outbound_ruleset/resource_genesyscloud_outbound_ruleset.go

```diff
@@ -114,12 +114,16 @@ func deleteOutboundRuleset(ctx context.Context, d *schema.ResourceData, meta int
 
 	return gcloud.WithRetries(ctx, 1800*time.Second, func() *resource.RetryError {
 		_, respCode, err := proxy.getOutboundRulesetById(ctx, d.Id())
+
+		//Now that I am checking for th error string of API 404 and there is no error, I need to move the isStatus404
+		//moved out of the code
+		if gcloud.IsStatus404ByInt(respCode) {
```

Comment:
I moved this out because once I started checking for the string in the message and not returning an err, there was not need to keep this outside the loop